### PR TITLE
[NUI] Reduce code duplication - clean up useless disposable codes

### PR DESCRIPTION
**Pull request by elishateng**
### Description of Change ###

1. Clean up useless disposable codes

### API Changes ###

 - ACR: http://suprem.sec.samsung.net/jira/browse/TCSACR-264

### CODE REVIEW

**Review by AdunFang — APPROVED** [review]
Seems OK.

**Comment by dongsug-song** [comment]
[TizenFX-API-Break-checking.zip](https://github.com/dalihub/TizenFX/files/3549621/TizenFX-API-Break-checking.zip)

Hello, Mr. Teng
please check the attached file above.
you can compare the API lists and there are many differences. 
there must be no difference, otherwise we need to process ACR as you know.
thank you.

**Comment by elishateng** [comment]
> [TizenFX-API-Break-checking.zip](https://github.com/dalihub/TizenFX/files/3549621/TizenFX-API-Break-checking.zip)
> 
> Hello, Mr. Teng
> please check the attached file above.
> you can compare the API lists and there are many differences.
> there must be no difference, otherwise we need to process ACR as you know.
> thank you.

Hello, Mr. SSong, I have attached the ACR link in Description. Please check it~

**Comment by dongsug-song** [comment]
Hello, Mr. Teng
please change the **http://suprem.sec.samsung.net/jira/browse/TCSACR-264** as below.
thank you.

- Remove => Deprecate (**remove** means "we delete it right now", but **deprecate** means "we will delete it later after maintaining 2 versions" for users to get some time to fix their codes)

- format =>
file path and name
[ACR action][member type] API name : brief reason
(example)
```
in Tizen.NUI/BaseComponents/Scrollable.cs
[Deprecate][Member Variable] public static readonly int OVERSHOOT_EFFECT_COLOR : not used anymore. can be replaced by XXX

```

**Comment by dongsug-song** [comment]
Hello, Mr Teng
I understand what this patch is for.
However, I am very sure there will be no build break when this patch is merged into VD p4v.
So, I suggest to make them all as hidden-api (public with [EditorBrowsable] and [Obsolete] attributes) , then there would be no build break.
**(example)**
`[EditorBrowsable(EditorBrowsableState.Never)]`
`[Obsolete("Deprecated in API6; Will be removed in API9.")]`
`public class Property : global::System.IDisposable`

ACR will be kept going and the hidden-api will be actually deleted after 2 version release of tizen.
How do you think about?

**Comment by elishateng** [comment]
> Hello, Mr Teng
> I understand what this patch is for.
> However, I am very sure there will be no build break when this patch is merged into VD p4v.
> So, I suggest to make them all as hidden-api (public with [EditorBrowsable] and [Obsolete] attributes) , then there would be no build break.
> **(example)**
> `[EditorBrowsable(EditorBrowsableState.Never)]`
> `[Obsolete("Deprecated in API6; Will be removed in API9.")]`
> `public class Property : global::System.IDisposable`
> 
> ACR will be kept going and the hidden-api will be actually deleted after 2 version release of tizen.
> How do you think about?

Hello, Mr. SSong,
Thank you for your remind, I'll update the patch.

File: src/Tizen.NUI/src/internal/Camera.cs [hunk 1]
```diff
@@ -64,142 +55,34 @@ protected override void Dispose(DisposeTypes type)
             base.Dispose(type);
         }
 
-
-
-        /// <since_tizen> 3 </since_tizen>
-        public class Property : global::System.IDisposable
+        internal class Property
```
Comment (dongsug-song) [inline src/Tizen.NUI/src/internal/Camera.cs]: please make this HiddenAPI (not internal)

File: src/Tizen.NUI/src/internal/LinearConstrainer.cs [hunk 2]
```diff
@@ -65,107 +55,17 @@ protected override void Dispose(DisposeTypes type)
             base.Dispose(type);
         }
 
-        /// <since_tizen> 3 </since_tizen>
-        public class Property : global::System.IDisposable
+        internal class Property
```
Comment (dongsug-song) [inline src/Tizen.NUI/src/internal/LinearConstrainer.cs]: please make this HiddenAPI (not internal) ; please add [EditorBrowsable], [Obsolete] attributes like others done.

File: src/Tizen.NUI/src/internal/Model3dView.cs [hunk 3]
```diff
@@ -66,116 +57,16 @@ protected override void Dispose(DisposeTypes type)
             base.Dispose(type);
         }
 
-        /// <since_tizen> 3 </since_tizen>
-        public new class Property : global::System.IDisposable
+        internal new class Property
```
Comment (dongsug-song) [inline src/Tizen.NUI/src/internal/Model3dView.cs]: please make this HiddenAPI (not internal) ; please add [EditorBrowsable], [Obsolete] attributes like others done.

File: src/Tizen.NUI/src/internal/PageTurnView.cs [hunk 4]
```diff
@@ -424,107 +411,11 @@ public static PageTurnView GetPageTurnViewFromPtr(global::System.IntPtr cPtr)
             return ret;
         }
 
-
-        /// <since_tizen> 3 </since_tizen>
-        public new class Property : global::System.IDisposable
+        internal new class Property
```
Comment (dongsug-song) [inline src/Tizen.NUI/src/internal/PageTurnView.cs]: please make this HiddenAPI (not internal) ; please add [EditorBrowsable], [Obsolete] attributes like others done.

File: src/Tizen.NUI/src/internal/PathConstrainer.cs [hunk 5]
```diff
@@ -65,105 +56,11 @@ protected override void Dispose(DisposeTypes type)
             base.Dispose(type);
         }
 
-
-
-        /// <since_tizen> 3 </since_tizen>
-        public class Property : global::System.IDisposable
+        internal class Property
```
Comment (dongsug-song) [inline src/Tizen.NUI/src/internal/PathConstrainer.cs]: please make this HiddenAPI (not internal) ; please add [EditorBrowsable], [Obsolete] attributes like others done.

File: src/Tizen.NUI/src/internal/RenderTask.cs [hunk 6]
```diff
@@ -73,108 +64,12 @@ public static RenderTask GetRenderTaskFromPtr(global::System.IntPtr cPtr)
             return ret;
         }
 
-        /// <since_tizen> 3 </since_tizen>
-        public class Property : global::System.IDisposable
+        internal class Property
```
Comment (dongsug-song) [inline src/Tizen.NUI/src/internal/RenderTask.cs]: please make this HiddenAPI (not internal) ; please add [EditorBrowsable], [Obsolete] attributes like others done.

**Pull request merged**